src/instructlab/configuration.py

```diff
@@ -310,6 +291,33 @@ def api_base(self):
         return get_api_base(self.host_port)
 
 
+class _generate(BaseModel):
+    """Class describing configuration of the generate sub-command."""
+
+    # model configuration
+    model_config = ConfigDict(extra="ignore")
+
+    # required fields
+    model: StrictStr
+    taxonomy_path: StrictStr
+    taxonomy_base: StrictStr
+
+    # additional fields with defaults
+    teacher: _serve = Field(default_factory=_serve)
+    num_cpus: PositiveInt = DEFAULTS.NUM_CPUS
+    chunk_word_count: PositiveInt = DEFAULTS.CHUNK_WORD_COUNT
+    # DEPRECATED: see sdg_scale_factor instead
+    # Left in place so that we can still detect and give a warning if its
+    # specified in an old configuration file.
+    num_instructions: Optional[int] = Field(
+        default=-1, deprecated="see 'sdg_scale_factor' instead", exclude=True
+    )
+    sdg_scale_factor: Optional[PositiveInt] = DEFAULTS.SDG_SCALE_FACTOR
+    output_dir: StrictStr = Field(default_factory=lambda: DEFAULTS.DATASETS_DIR)
```

Comment:
You can remove these `Field`s as they were previously used to prevent the config values from being cached before we ran tests. Nowadays, each test will reset the home directories to point to the [temporary path fixture](https://github.com/instructlab/instructlab/blob/c6adcb5ddd84d9358a2db56c5f30b237420f9b13/tests/conftest.py#L47).

```suggestion
    output_dir: StrictStr = DEFAULTS.DATASETS_DIR
```

Reply:
@RobotSail The entire `_generate` section was moved in this PR, and one addition was made (the addition of the `teacher` member. 

The line you mentioned is the same as what's in `main` here (https://github.com/instructlab/instructlab/blob/main/src/instructlab/configuration.py#L266). To keep the scope of this PR focused I'd prefer to open up a follow up PR with that change, and possibly changes to any other variables in this section.

Reply:
Makes sense, let's approve this then and then fix it later.